### DISCUSSION TITLE
Ability to run a saved query from the data entry page

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5622] [Feature Requests]
<img width="392" alt="image" src="https://user-images.githubusercontent.com/37256050/186797405-a9571dcb-ab41-4e6d-ae5d-a7d189f05f2e.png">

Just like in the first screenshot, it would be nice to be able to run a pre-selected query automatically when clicking the 👁️ icon.

**Requested By:** [**CSIRO**](https://app.asana.com/0/1201972579963460/1202844554923815/f)

**Comment by maxpatiiuk** [comment]
How about a user preference that automatically runs the query as soon as it is opened?

**Comment by danb213** [comment]
That would be a good solution - would it be a global setting for the user or a per-query feature? I think a global setting for the user would work well

**Comment by maxpatiiuk** [comment]
See also https://github.com/specify/specify7/issues/2047#issuecomment-1235511827

**Comment by danb213** [comment]
I just realised that clicking on the query name takes you to the same page that the edit button does, perhaps that could be the link to running the query automatically (so you don't need to add an icon to this popup)?

**Comment by maxpatiiuk** [comment]
Hmm, not sure what you mean.
Pressing the pencil button in the list of queries brings up a dialog to rename the query.
Clicking on the query name opens the query definition.

Adding a user preference like mentioned here earlier to auto run queries on open sounds like an good solution (and is easy to implement)

**Comment by danb213** [comment]
Sorry I was mistaken about what happens when you click on the pencil (not sure what I was thinking there!).

The user preference option would work well for us
